tools/file_packager.py

```diff
@@ -650,20 +673,22 @@ def generate_js(data_target, data_files, metadata):
           code += ('''Module['FS_createPath'](%s, %s, true, true);\n'''
                    % (json.dumps('/' + '/'.join(parts[:i])), json.dumps(parts[i])))
           partial_dirs.append(partial)
-
-  if options.has_preloaded:
+  has_preloaded = any(f.mode == 'preload' for f in data_files)
+  has_embedded = any(f.mode == 'embed' for f in data_files)
+  if has_preloaded:
     # Bundle all datafiles into one archive. Avoids doing lots of simultaneous
     # XHRs which has overhead.
     start = 0
     with open(data_target, 'wb') as data:
-      for file_ in data_files:
-        file_.data_start = start
-        curr = utils.read_binary(file_.srcpath)
-        file_.data_end = start + len(curr)
-        if AV_WORKAROUND:
-            curr += '\x00'
-        start += len(curr)
-        data.write(curr)
+      if file_.mode == 'preload':
```

Comment:
I guess the old code assume that either all files were preloads or embedded, and not a combination of both?

Do you want to be able to do both at the same time?  I think maybe we should just make that invalid?

Reply:
maybe that warrants a separate PR since that is a breaking change. I will leave that to the maintainer to introduce the breaking change

Reply:
Absolutely, yes.  I'm looking into it now.